docs/introduction/index.md

```diff
@@ -0,0 +1,226 @@
+                 .setName(e.getName())
+    }
+
+    @Subscribe
+    void on(TaskCompleted e, EventContext ctx) {
+        builder().setWhenDone(ctx.getTimestamp());
+    }
+}</code></pre>
+
+#### Repositories
+The framework provide default implementations for repositories.
+A custom `Repository` class may be needed for:
+  * <strong>Dispatching messages to entities in a non-standard way</strong>.
+    By default, a command is dispatched using the first field of the command message.
+    An event is dispatched by the ID of the entity which generated the event.
+  * <strong>Domain-specific operations</strong> on entities of this kind.
+  
+Repositories are added to the Bounded Context they belong when it is created:
+
+<pre class="highlight lang-java">
+<code>BoundedContext tasksContext = BoundedContext.multiTenant("Tasks")
+    .add(TaskAggregate.class) // use default repository impl.
+    .add(new TaskProjectionRepository())
+    .build();
+</code></pre>  
+
+This wires repositories into the message delivery mechanism of corresponding
+[Buses](concepts.html#message-buses).
+  
+#### Testing
+Implementation of the Bounded Context is tested using the messaging paradigm.
+The following code snippet asserts that handling a command `CreateTask` produces one 
+`TaskCreated` event with expected arguments.
+ 
+<pre class="highlight lang-java"><code>// Given
+BlackBoxBoundedContext context = BlackBoxBoundedContext.from(tasksContext);
+
+// When
+context.receivesCommand(createTask());
+
+// Then
+TaskCreated expected = TaskCreated.newBuilder()
+    .setId(id)
+    .setName(name)
+    .build();
+
+EventSubject assertEvents = 
+    context.assertEvents()
+           .withType(TaskCreated.class)    
+
+assertEvents.hasSize(1);
+assertEvents.message(0)
+       .comparingExpectedFieldsOnly()
+       .isEqualTo(expected);  
+</code></pre>
+
+Modification of entities is also tested. The following code snippet asserts that the state
+of the `TaskAggregate` was also updated with expected arguments.
+
+<pre class="highlight lang-java"><code>EntitySubject assertEntity = 
+    context.assertEntityWithState(Task.class, id);
+Task expectedState = Task.newBuilder()
+    .setId(id)
+    .setName(name)
+    .build();
+assertEntity.hasStateThat()
+    .comparingExpectedFieldsOnly()
+    .isEqualTo(expectedState);
+</code></pre>
+
+## Deployment
+
+### Configuring Server Environment
+
+### Assembling Application
+
+The server-side application is composed with its Bounded Contexts.
+
+<pre class="highlight lang-java"><code>Server server = Server.atPort(portNumber)
+    .add(tasksContext)
+    .add(usersContext)
+    .add(commentsContext)
+    .build();
+server.start();    
+</code></pre>
+
+This exposes [`CommandService`](concepts.html#command-service), 
+[`QueryService`](concepts.html#query-service), and 
+[`SubscriptionService`](concepts.md#subscription-service) for client-side connections.
+
+## Repeating the cycle
+
+## Client application development
+ 
+
```

Comment:
Please kill the empty lines.

I would also append something like `This section is under development. For now, you may refer to the Javadoc of the Java client API or web client API.` — and put the links to the corresponding repositories (`core-java/client` and `web`).